### PR TITLE
[Documentation] Add instruction how to execute built-in commands of various shells.

### LINKED ISSUE #26429: Unable to run built-in commands using OS.execute
[Sample Project](https://github.com/godotengine/godot/files/2918084/test_execute.zip)

**Godot version:**
3.0.6
8d117b214f2bcd14015532fdfbed9b3f059da0e6

**OS/device including version:**
Windows 10 x64

**Issue description:**

```
var output = []
OS.execute('echo', ['somethingyo'], true, output)
```

Produces the following:

```
'"echo"' is not recognized as an internal or external command,
operable program or batch file.
```

It appears that the `path` argument is being automatically surrounded in quotes, per [OS_Windows::execute](https://github.com/godotengine/godot/blob/master/platform/windows/os_windows.cpp#L2450).

As mentioned in [this issue](https://github.com/godotengine/godot/issues/10172), this can currently be circumvented by calling `CMD /K "<command>"` - however this is not an elegant solution as it requires launching a second process. I propose one of the following solutions which doesn't require this workaround.

Solution 1:
----------

Remove automatic quotes. This would have the downside of requiring the user to use quotes 
when necessary.

Solution 2:
----------

Enclose the path with quotes only when there is whitespace in the path.

Solution 3:
----------

Check if the command is a DOS command, and enclose the path in quotes if it is not.

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes #26429

### CODE REVIEW

File: doc/classes/OS.xml [hunk 1]
```diff
@@ -140,6 +140,9 @@
 				[/csharp]
 				[/codeblocks]
 				[b]Note:[/b] This method is implemented on Android, iOS, Linux, macOS and Windows.
+				[b]Note:[/b] To execute the Windows command interpreter built-in command, specify [code]cmd.exe[/code] in [code]path[/code], [code]/c[/code] as the first argument, and desired command as the second argument.
+				[b]Note:[/b] To execute the PowerShell built-in command, specify [code]powershell.exe[/code] in [code]path[/code], [code]-Command[/code] as the first argument, and desired command as the second argument.
+				[b]Note:[/b] To execute the Unix shell built-in command, specify shell executable name in [code]path[/code], [code]-c[/code] as the first argument, and desired command as the second argument.
```
Comment (akien-mga) [inline doc/classes/OS.xml:145]:
```suggestion
				[b]Note:[/b] To execute a Windows command interpreter built-in command, specify [code]cmd.exe[/code] in [code]path[/code], [code]/c[/code] as the first argument, and the desired command as the second argument.
				[b]Note:[/b] To execute a PowerShell built-in command, specify [code]powershell.exe[/code] in [code]path[/code], [code]-Command[/code] as the first argument, and the desired command as the second argument.
				[b]Note:[/b] To execute a Unix shell built-in command, specify shell executable name in [code]path[/code], [code]-c[/code] as the first argument, and the desired command as the second argument.
```

**Review by akien-mga — APPROVED** [review]

File: doc/classes/OS.xml (hunk 1 above)
Comment (bruvzg) [reply doc/classes/OS.xml]: Fixed.

**Review by mhilbrunner — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.3.

**Pull request merged**